client/signup/steps/site-style/index.jsx

```diff
@@ -0,0 +1,181 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import React, { Component } from 'react';
+import i18n, { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { find } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import Button from 'components/button';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormRadio from 'components/forms/form-radio';
+import StepWrapper from 'signup/step-wrapper';
+import SignupActions from 'lib/signup/actions';
+import { setSiteStyle } from 'state/signup/steps/site-style/actions';
+import { getSiteStyle } from 'state/signup/steps/site-style/selectors';
+import { getSiteType } from 'state/signup/steps/site-type/selectors';
+import { getSiteStyleOptions } from 'lib/signup/site-styles';
+import { getSignupStepsSiteTopic } from 'state/signup/steps/site-topic/selectors';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export class SiteStyleStep extends Component {
+	static propTypes = {
+		flowName: PropTypes.string,
+		goToNextStep: PropTypes.func.isRequired,
+		positionInFlow: PropTypes.number,
+		submitSiteStyle: PropTypes.func.isRequired,
+		signupProgress: PropTypes.array,
+		styleOptions: PropTypes.array.isRequired,
+		stepName: PropTypes.string,
+		siteStyle: PropTypes.string,
+		siteType: PropTypes.string,
+		translate: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+		const selectedStyle =
+			find( props.styleOptions, [ 'id', props.siteStyle ] ) || props.styleOptions[ 0 ];
+		this.state = {
+			siteStyle: selectedStyle.id,
+			themeSlugWithRepo: selectedStyle.theme,
+		};
+	}
+
+	componentDidMount() {
+		SignupActions.saveSignupStep( {
+			stepName: this.props.stepName,
+		} );
+	}
+
+	handleStyleOptionChange = event => {
+		const selectedStyle = find( this.props.styleOptions, [ 'id', event.currentTarget.value ] );
+		this.setState(
+			{
+				siteStyle: selectedStyle.id,
+				themeSlugWithRepo: selectedStyle.theme,
+			},
+			() => this.props.setSiteStyle( selectedStyle.id )
+		);
+	};
+
+	handleSubmit = event => {
+		event.preventDefault();
+		this.props.submitSiteStyle( this.state );
+	};
+
+	renderStyleOptions() {
+		return this.props.styleOptions.map( siteStyleProperties => {
+			const isChecked = siteStyleProperties.id === this.state.siteStyle;
```

Comment:
Since `this.state.siteStyle` and `this.props.siteStyle` are identical, we could use the prop here if we removed the state.

Reply:
We just have to be careful to declare a default should the site style value be unavailable. 👍